### ISSUE TITLE
Explicit behaviour on warning and errors

### ISSUE THREAD

**Issue by ghost** [MobilityData/gtfs-validator#60]
**Is your feature request related to a problem? Please describe.**
Right now, the validator fills errors and warning lists but does not act on it at all.

**Describe the solution you'd like**
This task consist in exploring how we are gonna code handling of warning and errors. So far, we have the following guidelines
- errors should provoke an orderly stop of the validation process with explicit user output
- unexpected enum values should generate warnings
-consider adding a new command line option to treat warnings as error (or the other way around)

**Additional context**
Relates to #47

**Comment by lionel-nj** [comment]
I see 2 different approches:

1. Check for warnings before the execution of a new use case

- useCase A (point of entry)
  - execution + generation of notices

<-- check for warning as a precondition for useCase B-->

- useCase B (second use case)
  - execution + generation of notices

<-- check for warning as a precondition for useCase B-->

- useCase C (third use case)
  - execution + generation of notices

 In this scenario, if a warning was found in the notices generated by useCase A, we stop the process and do not execute useCase B.

2. Constant check for warning 

```
while(noWarningWasFound) do:
  execute useCase A
  execute useCase B
  ...
```

In this scenario, the process is stopped as soon as a warning is found.
The boolean `noWarningWasFound` would be updated at the end of each use case, after checking that no warning was found in the validation result repository.

`noWarningWasFound` could be defined as `validationResultRepository.size() == 0`
 

Both solutions require logging both the start and the end of execution of a use case to be able to retrace the operation that led to the warning.


As far as unexpected enum values are concerned, I suggest to generate a warning as soon as an unexpected value is passed. Therefore, with one of the previous mechanism  (1 or 2), the validation process would stop. 
For example( in file https://github.com/MobilityData/gtfs-validator/blob/stops-classes/domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/stops/WheelchairBoarding.java from PR#44), instead of having this:

```
    static public WheelchairBoarding fromInt(Integer fromValue) {
        if (fromValue == null) {
            return UNKNOWN_WHEELCHAIR_BOARDING;
        }
        return Stream.of(WheelchairBoarding.values())
                .filter(enumItem -> enumItem.value == fromValue)
                .findAny()
                .orElse(**UNKNOWN_WHEELCHAIR_BOARDING**);
    }
```

we could implement something like that: 

```
    @SuppressWarnings("OptionalGetWithoutIsPresent")
    static public WheelchairBoarding fromInt(Integer fromValue) throws Exception {
        if (fromValue == null) {
            return UNKNOWN_WHEELCHAIR_BOARDING;
        }
        try {
            WheelchairBoarding toReturn = Stream.of(WheelchairBoarding.values())
                    .filter(enumItem -> enumItem.value == fromValue)
                    .findAny()
                    .get();
            return toReturn;
        } catch (Exception e) {
            throw new Exception("Unexpected enum value");
        }
    }
```

The main issue I see with this solution is the fact that we directly throw an exception, instead of generating a warning that  will lead to stop the validation process.

**Comment by barbeau** [comment]
For enums, IMHO we should treat them like any other error or warnings. This will give us flexibility to easily change how they are handled later. For example, for simplicity now we've decided to flag these as errors, but eventually I believe we'll need to change these to warnings if we want to be able to validate all legitimate GTFS datasets.

Do we really need to abort after each use case? It seems like there would be natural stopping points within the application where we can't move to the next stage if there are errors. For example - after all type validation for all files, or before constructing stop_times entities if there are errors with stops.

I'd suggest we allow execution until we reach one of those stopping points, and then check to see if any errors have been generated, and if so stop execution then. This will also help agencies communicate more efficiently with their vendors if they are running it on behalf of the vendor - they can provide a list of more errors at a time (vs. one at a time).

**Comment by lionel-nj** [comment]
"unexpected enum values should generate warnings
", at some point this issue is linked to https://github.com/MobilityData/gtfs-validator/issues/48

I was thinking of using genetic types for the enums to be able to have an single implementation of fromInt method. This method fromInt would return ERROR(-1) as an enum value.  The next step would be to handle this enum value.

If that works it could solve these two issues. However, given the requirements of the specification (sometimes empty enums have a default value, in other cases the field can not be empty), this solution might not be optimal.

**Comment by barbeau** [comment]
Closing - on errors we fail if "abort on error" command-line option is enabled (default) and if it's not enabled then further behavior after the first error may be unpredictable (further downstream errors may be generated). On warnings we never abort.
